lib/liquid/c.rb

```diff
@@ -73,34 +69,42 @@ def new_block_body
     end
   end
 
+  alias_method :ruby_new_tokenizer, :new_tokenizer
+
+  def new_tokenizer(source, line_number: nil, for_liquid_tag: false)
+    unless liquid_c_nodes_disabled?
+      source = source.to_s
+      if source.bytesize <= Liquid::C::Tokenizer::MAX_SOURCE_BYTE_SIZE
+        source = source.encode(Encoding::UTF_8)
+        return Liquid::C::Tokenizer.new(source, line_number || 0, for_liquid_tag)
```

Comment:
Shouldn't it be `|| 1`?

Reply:
The C code uses `0` to mean that line numbers aren't tracked, similar to if `line_number` is `nil` in the ruby tokenizer.